cdap-master/src/main/java/co/cask/cdap/data/runtime/main/MasterServiceMain.java

```diff
@@ -109,204 +115,82 @@
-      new TwillModule(),
-      new DiscoveryRuntimeModule().getDistributedModules(),
-      new AppFabricServiceRuntimeModule().getDistributedModules(),
-      new ProgramRunnerRuntimeModule().getDistributedModules(),
-      new DataSetServiceModules().getDistributedModules(),
-      new DataFabricModules().getDistributedModules(),
-      new DataSetsModules().getDistributedModules(),
-      new MetricsClientRuntimeModule().getDistributedModules(),
-      new ServiceStoreModules().getDistributedModule(),
-      new ExploreClientModule(),
-      new NotificationFeedServiceRuntimeModule().getDistributedModules(),
-      new NotificationServiceRuntimeModule().getDistributedModules(),
-      new StreamAdminModules().getDistributedModules()
-    );
-
-    // Initialize ZK client
-    zkClientService = baseInjector.getInstance(ZKClientService.class);
-    kafkaClientService = baseInjector.getInstance(KafkaClientService.class);
-    metricsCollectionService = baseInjector.getInstance(MetricsCollectionService.class);
-    dsService = baseInjector.getInstance(DatasetService.class);
-    serviceStore = baseInjector.getInstance(ServiceStore.class);
-    exploreClient = baseInjector.getInstance(ExploreClient.class);
-    secureStoreUpdater = baseInjector.getInstance(TokenSecureStoreUpdater.class);
+    cleanupTempDir();
 
-    // this should probably be done in NamespaceService#init()
-    createSystemHBaseNamespace();
-    checkTransactionRequirements();
     checkExploreRequirements();
-  }
-
-  private void createSystemHBaseNamespace() {
-    HBaseTableUtil tableUtil = baseInjector.getInstance(HBaseTableUtil.class);
-    try {
-      HBaseAdmin admin = new HBaseAdmin(hConf);
-      tableUtil.createNamespaceIfNotExists(admin, Constants.SYSTEM_NAMESPACE_ID);
-    } catch (MasterNotRunningException e) {
-      Throwables.propagate(e);
-    } catch (ZooKeeperConnectionException e) {
-      Throwables.propagate(e);
-    } catch (IOException e) {
-      Throwables.propagate(e);
-    }
-  }
-
-  /**
-   * The transaction coprocessors (0.94 and 0.96 versions of {@code DefaultTransactionProcessor}) need access
-   * to CConfiguration values in order to load transaction snapshots for data cleanup.
-   */
-  private void checkTransactionRequirements() {
-    try {
-      new ConfigurationTable(hConf).write(ConfigurationTable.Type.DEFAULT, cConf);
-    } catch (IOException ioe) {
-      throw Throwables.propagate(ioe);
-    }
-  }
-
-  /**
-   * Check that if Explore is enabled, the correct jars are present on master node,
-   * and that the distribution of Hive is supported.
-   */
-  private void checkExploreRequirements() {
-    if (!exploreEnabled) {
-      return;
-    }
-
-    // This check will throw an exception if Hive is not present or if it's distribution is unsupported
-    ExploreServiceUtils.checkHiveSupport();
+    login();
   }
 
   @Override
   public void start() {
+    // this should probably be done in NamespaceService#init()
```

Comment:
Yes :) When I had added that, there was a thought of creating a ``NamespaceService``. But I don't think we need that anymore.